### DISCUSSION TITLE
kens-open test

### DISCUSSION THREAD

**Discussion by crossedfinger-j** [ANLAB-KAIST/KENSv3#181] [Project Q&A 2. Transport Layer (TCP)]
질문이 있습니다. 
과제 설명에 보면, Using connect() or close(), you need a timer in order not to wait indefinitely. Check out the
following link to use timers. 이라는 부분이 있습니다. 따라서, close systemcall 에서도 4-way handshake 부분을 적용해야할 것 같아서 코드 작성 후 kens-test-open에 관한 파일을 열어보았는데 소켓을 연 후, 바로 소켓을 닫는 것 같습니다. 2번 과제에서 close 부분에 4-way handshake를 적용해야 하는지 과제에 대해 질문드립니다. 

I have a question.
As the pdf of PA 2, "Using connect() or close(), you need a timer in order not to wait indefinitely. Check out the
There is a part called following link to use timers". Therefore, I think we need to apply the 4-way handshake part in the close system call as well, so I opened the file about the kens-test-open after writing the code, and it seems that the socket is closed right after opening the socket. In task 2, I am asking about the task of applying the 4-way handshake to the close part.

**Comment by Pusnow** [comment]
For the `kens-test-open`, you don't, but for other test cases, you should.
